src/bam_to_bed/intron_retention.py

```diff
@@ -0,0 +1,163 @@
+import os
+import sys
+import time
+import pysam
+import string
+import random
+import argparse
+from functools import partial
+from multiprocessing import Pool
+
+
+###############################################################################################################################################################
+# CURRENT INSTRUCTIONS
+#
+# To obtain properly formatted coordinate sorted indexed gene model reference BED file
+# that includes only introns, do the following actions with your AltAnalyze formatted
+# Hs_Ensembl_exon.txt (for example) file.
+#
+# 1. Make sure you have these columns in you input Hs_Ensembl_exon.txt file
+#    head -n 1 Hs_Ensembl_exon.txt
+#    gene	exon-id	chromosome	strand	exon-region-start(s)	exon-region-stop(s)	constitutive_call	ens_exon_ids	splice_events	splice_junctions
+#
+# 2. Convert to BED format excluding all not intron records
+#    cat Hs_Ensembl_exon.txt | grep -v "gene" | awk '$2 ~ /^I/ {print $3"\t"$5"\t"$6"\t"$2"-"$1"\t"0"\t"$4}' | sort -k1,1 -k2,2n -k3,3n | bgzip > hs_ref.bed.gz
+#    tabix -p bed hs_ref.bed.gz
+###############################################################################################################################################################
+
+
+#################################################################################################
+# DEPRECATED INSTRUCTIONS
+# keep it in case we decide to support GTF input in future
+#
+# (grep ^"#" input.gtf; grep -v ^"#" input.gtf | sort -k1,1 -k4,4n) | bgzip > input.sorted.gtf.gz
+# tabix -p gff input.sorted.gtf.gz
+#################################################################################################
+
+
+def is_paired(args):

```

Comment:
I think we should change the logic of this method.  Right now, it examines 20 reads and returns False if **any** of them do not have a mate or the mate is unmapped.  It also seeks all over the file, because `mate()` actually updates the iterator to point at the mate read.

I think that if a file represents paired data, all reads in the files will be paired.  I think it would be sufficient to do something like this:

```
with pysam.AlignmentFile(args.bam, mode="rb", threads=args.threads) as bam_handler:
         read = next(bam_handler.fetch())
         return read.is_paired()
```

Reply:
Agree, updated